packages/test/test-service-load/src/runner.ts

```diff
@@ -242,14 +283,80 @@ async function runnerProcess(
                 await baseLogger.flush({ url, runId: runConfig.runId });
             }
         }
-        return 0;
+
+        const nextFactoryPermutation2 = iterator.next();
+        if (nextFactoryPermutation2.done === true) {
+            throw new Error("Factory permutation iterator is expected to cycle forever");
+        }
+        // const values = nextFactoryPermutation2.value;
+        // const container2 = await loadContainer(values.documentServiceFactory, values.headers);
+        // container2.connect();
+        // const test2 = await requestFluidObject<ILoadTest>(container2, "/");
+
+        let verified: boolean | undefined;
+        while (verified === undefined) {
+            const values = nextFactoryPermutation2.value;
+            const verifyContainer = await loadContainer(values.documentServiceFactory, values.headers);
+            verifyContainer.connect();
+            const verifyTest = await requestFluidObject<ILoadTest>(verifyContainer, "/");
+            try {
+                verified = await verifyTest.verify(runConfig, logger, runContext);
+            } catch (e) {
+                printStatus(runConfig, "error while verifying:");
+                console.log(e);
+            } finally {
+                if (!verifyContainer.closed) {
+                    verifyContainer.close();
+                }
+            }
+        }
+
+        printStatus(runConfig, verified ? "verified" : `not verified: ${runContext.fail}`);
+        return verified ? 0 : -1;
     } catch (e) {
         printStatus(runConfig, `error: loading test`);
         console.error(e);
         return -1;
     }
 }
 
+async function scheduleOffline(
+    dsf: FaultInjectionDocumentServiceFactory,
+    container: IContainer,
+    runConfig: IRunConfig,
+): Promise<string | undefined> {
+    // should it be possible to go offline without closing?
+    if (container.connectionState !== ConnectionState.Connected && !container.closed) {
+        await Promise.race([
+            new Promise((res) => container.once("connected", res)),
+            new Promise((res) => container.once("closed", res)),
+        ]);
+    }
+
+    if (container.closed) {
+        return;
+    }
+
+    // TODO: config values
+    const injectionTime = random.integer(0, 30_000)(runConfig.randEng);
+    const offlineTime = random.integer(5_000, 15_000)(runConfig.randEng);
+
+    await new Promise<void>((res) => setTimeout(res, injectionTime));
+    assert(container.resolvedUrl !== undefined, "no url");
+
+    const ds = dsf.documentServices.get(container.resolvedUrl);
+    assert(!!ds, "no documentServices");
+
+    printStatus(runConfig, `going offline! closing in ${offlineTime / 1000} seconds!`);
+    ds.goOffline();
+
+    await new Promise<void>((res) => setTimeout(res, offlineTime));
+    if (!container.closed) {
+        printStatus(runConfig, "stashing changes!");
+        return container.closeAndGetPendingLocalState();
```

Comment:
closeAndGetPendingLocalState could be called even if it is not offline, right?

Reply:
Yes, but we would only stash what happens to be pending at that time.